src/file.ts

```diff
@@ -2294,21 +2330,61 @@ class File extends ServiceObject<File> {
     const expiresInSeconds =
         Math.round(expiresInMSeconds / 1000);  // The API expects seconds.
 
-    const config: GetSignedUrlConfigInternal = Object.assign({}, cfg);
-
-    config.action = ({
-      read: 'GET',
-      write: 'PUT',
-      delete: 'DELETE',
-      resumable: 'POST',
-    } as {[index: string]: string})[config.action];
+    const method = ActionToHTTPMethod[cfg.action];
 
     const name = encodeURIComponent(this.name);
-    config.resource = '/' + this.bucket.name + '/' + name;
+    const resource = `/${this.bucket.name}/${name}`;
```

Comment:
The new conformance tests will expect the trailing slash to be removed when file name is not given. (List objects request).

Reply:
List objects would be a bucket level signed URL - see below :)

Reply:
We've decided to tackle Bucket level SignedURL in a future PR.